frontend/src/components/CsetComparisonPage.jsx

```diff
@@ -62,17 +62,26 @@ import {LI} from './AboutPage';
 // TODO: Color table: I guess would need to see if could pass extra values/props and see if table widget can use that
 //  ...for coloration, since we want certain rows grouped together
 export async function fetchGraphData(props) {
-  let {dataGetter, comparison_pair, codeset_ids, cids, newCset = {}} = props;
+  let {dataGetter, compareOpt, codeset_ids, cids, newCset = {}} = props;
   let promises = [ // these can run immediately
     dataGetter.fetchAndCacheItems(dataGetter.apiCalls.cset_members_items,
         codeset_ids),
     dataGetter.fetchAndCacheItems(dataGetter.apiCalls.csets, codeset_ids),
   ];
   let comparison_rpt;
-  if (comparison_pair) {
-    comparison_rpt = dataGetter.axiosCall(
-        `single-n3c-comparison-rpt?pair=${comparison_pair}`,
-        {sendAlert: false, skipApiGroup: true});
+  if (compareOpt) {
+    if (codeset_ids.length !== 2) {
+      throw new Error("Can't use the compare option unless there are exactly two codeset_ids");
+    }
+    if (compareOpt === 'compare-precalculated') {
```

Comment:
I suppose this is the essential bugfix? The changes related to `compare-precalculated`